plugins/browser-plugin-event-specifications/src/index.ts

```diff
@@ -0,0 +1,43 @@
+import { BrowserPlugin, BrowserTracker } from '@snowplow/browser-tracker-core';
+import { integrations, AvailablePlugins } from './integrations';
+import { createEventSpecificationContext } from './utils';
+
+const _trackers: Record<string, BrowserTracker> = {};
+
+type EventSpecificationsPluginOptions = {
+  [k in AvailablePlugins]?: Record<string, string>;
+};
+
+const defaultPluginOptions = {};
+
+export function EventSpecificationsPlugin(
+  pluginOptions: EventSpecificationsPluginOptions = defaultPluginOptions
+): BrowserPlugin {
+  const options = { ...defaultPluginOptions, ...pluginOptions };
+  const enabledIntegrations = Object.keys(integrations).filter(
+    (integration) => options[integration as AvailablePlugins]
+  );
+  let trackerId: string;
+  return {
+    activateBrowserPlugin: (tracker) => {
+      trackerId = tracker.id;
+      _trackers[trackerId] = tracker;
+      return;
+    },
+    beforeTrack(payloadBuilder) {
+      enabledIntegrations.forEach((integrationKey) => {
+        const matchedEvent = integrations[integrationKey as AvailablePlugins].detectMatchingEvent(payloadBuilder);
+        if (!matchedEvent) {
+          return;
+        }
+        const matchingPluginEventMap = Object.values(options).find(
```

Comment:
Why do we consider all integrations here? For instance if the user configured the plugin like this:

```ts
EventSpecificationPlugin({
  OtherPlugin: { play_event: "wrong" },
  SnowplowMediaPlugin: { play_event: "wrong" },
})
```

And the current event was matched against the media plugin, wouldn't this code result in the `wrong` ES ID being taken?

This can't happen now since we only have one integration, but if there is such a problem, it would be good to solve it before we add more integrations.

Reply:
Yeap, fixed:

```ts
const matchingPluginEventMap = options[integrationKey as AvailablePlugins];
const matchingEventSpecificationId = matchingPluginEventMap && matchingPluginEventMap[matchedEvent];
if (!matchingEventSpecificationId) {
    return;
}
```

Plus both integrations and options can only have `AvailablePlugins` as keys

Reply:
looks good!